docs/source/get_involved/contributing.rst

```diff
@@ -0,0 +1,477 @@
+
+1. To install extra requirements from the root directory, run:
+
+   .. code:: bash
+
+      pip install -r docs/requirements.txt
+
+2. To build the website locally, run:
+
+   .. code:: bash
+
+      make docs
+
+You can find the generated files in the ``sktime/docs/_build/`` folder.
+To view the website, open ``sktime/docs/_build/html/index.html`` with
+your preferred web browser.
+
+Dependencies
+------------
+
+We try to keep the number of core dependencies to a minimum and rely on
+other packages as soft dependencies when feasible.
+
+.. note::
+
+   A soft dependency is a dependency that is only required to import
+   certain modules, but not necessary to use most functionality. A soft
+   dependency is not installed automatically when the package is
+   installed. Instead, users need to install it manually if they want to
+   use a module that requires a soft dependency.
+
+If you add a new dependency or change the version of an existing one,
+you need to update the following files:
+
+-  `sktime/setup.py <https://github.com/alan-turing-institute/sktime/blob/main/setup.py>`__
+   for package installation and minimum version requirements,
+-  `build_tools/requirements.txt <https://github.com/alan-turing-institute/sktime/blob/main/build_tools/requirements.txt>`__
+   for continuous integration and distribution,
+-  `docs/requirements.txt <https://github.com/alan-turing-institute/sktime/blob/main/docs/requirements.txt>`__
+   for building the documentation,
+-  `.binder/requirements.txt <https://github.com/alan-turing-institute/sktime/blob/main/.binder/requirements.txt>`__
+   for launching notebooks on Binder.
+
+If a user is missing a soft dependency, we raise a user-friendly error message.
+This is handled through our ``_check_soft_dependencies`` defined
+`here <https://github.com/alan-turing-institute/sktime/blob/main/sktime/utils/validation/_dependencies.py>`__.
+
+We use contiunous integration tests to check if all soft
+dependencies are properly isolated to specific modules.
+If you add a new soft dependency, make sure to add it
+`here <https://github.com/alan-turing-institute/sktime/blob/main/build_tools/azure/check_soft_dependencies.py>`__
+together with the module that depends on it.
+
+Coding style
+------------
+
+We follow the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`__
+coding guidelines. A good example can be found
+`here <https://gist.github.com/nateGeorge/5455d2c57fb33c1ae04706f2dc4fee01>`__.
+
+We use the `pre-commit <#Code-quality-checks>`_ workflow together with
+`black <https://black.readthedocs.io/en/stable/>`__ and
+`flake8 <https://flake8.pycqa.org/en/latest/>`__ to automatically apply
+consistent formatting and check whether your contribution complies with
+the PEP8 style.
+
+For docstrings, we use the `numpy docstring
+standard <https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard>`__.
```

Comment:
I'd add on to the sentence : "along with sktime specific conventions described in our :ref:`developer guide's documentation section <developer_guide_documentation>`

Reply:
Added the sentence